src/LoggerServiceFactory.php

```diff
@@ -34,9 +44,108 @@ public function __invoke(ContainerInterface $container, $name, ?array $options =
+                || ! $config['processor_plugin_manager'] instanceof AbstractPluginManager)
+            && $services->has('LogProcessorManager')
+        ) {
+            $config['processor_plugin_manager'] = $services->get('LogProcessorManager');
+        }
+
+        if (! isset($config['writers'])) {
+            return;
+        }
+
+        foreach ($config['writers'] as $index => $writerConfig) {
+            if (
+                isset($writerConfig['name'])
+                && ('db' === $writerConfig['name']
+                    || Writer\Db::class === $writerConfig['name']
+                    || 'laminaslogwriterdb' === $writerConfig['name']
+                )
+                && isset($writerConfig['options']['db'])
+                && is_string($writerConfig['options']['db'])
+                && $services->has($writerConfig['options']['db'])
+            ) {
+                // Retrieve the DB service from the service locator, and
+                // inject it into the configuration.
+                $db                                         = $services->get($writerConfig['options']['db']);
+                $config['writers'][$index]['options']['db'] = $db;
+                continue;
+            }
+
+            if (
+                isset($writerConfig['name'])
+                && ('mongo' === $writerConfig['name']
+                    || Writer\Mongo::class === $writerConfig['name']
+                    || 'laminaslogwritermongo' === $writerConfig['name']
+                )
+                && isset($writerConfig['options']['mongo'])
+                && is_string($writerConfig['options']['mongo'])
+                && $services->has($writerConfig['options']['mongo'])
+            ) {
+                // Retrieve the Mongo service from the service locator, and
+                // inject it into the configuration.
+                $mongoClient                                   = $services->get($writerConfig['options']['mongo']);
+                $config['writers'][$index]['options']['mongo'] = $mongoClient;
+                continue;
+            }
+
+            if (
+                isset($writerConfig['name'])
+                && ('mongodb' === $writerConfig['name']
+                    || Writer\MongoDB::class === $writerConfig['name']
+                    || 'laminaslogwritermongodb' === $writerConfig['name']
+                )
+                && isset($writerConfig['options']['manager'])
+                && is_string($writerConfig['options']['manager'])
+                && $services->has($writerConfig['options']['manager'])
+            ) {
+                // Retrieve the MongoDB Manager service from the service locator, and
+                // inject it into the configuration.
+                $manager                                         = $services->get($writerConfig['options']['manager']);
+                $config['writers'][$index]['options']['manager'] = $manager;
+            }
```

Comment:
We should add a `continue` in the previous conditional, so that if we add additional conditions in this loop, we don't accidentally move on to them after this one matches.